### DISCUSSION TITLE
New feature: using pretrained Flexynesis models in inference mode

### DISCUSSION THREAD

**Discussion by borauyar** [BIMSBbioinfo/flexynesis#126] [Announcements]
We’ve added support for inference-only mode in Flexynesis available from version `1.1.6`. 
Thanks a lot to @amitpande74 for developing this long-awaited feature. 

You can now pretrain a model once and reuse it to predict outcomes on a new test dataset. 
!! Warning: in the current version, the test dataset must contain the same omics modalities and features used during training.

Here is a short snippet on how to use it on inference mode:

```
# Download sample dataset
curl -L -o dataset1.tgz https://bimsbstatic.mdc-berlin.de/akalin/buyar/flexynesis-benchmark-datasets/dataset1.tgz 
tar -xzvf dataset1.tgz

# Train and save model
flexynesis \
  --data_path dataset1 \
  --model_class DirectPred \
  --target_variables Erlotinib \
  --fusion_type early \
  --hpo_iter 1 \
  --data_types gex,cnv \
  --outdir . \
  --prefix erlotinib 

# Inference on test dataset 
flexynesis \
  --pretrained_model erlotinib.final_model.pth \
  --artifacts erlotinib.artifacts.joblib \
  --data_path_test dataset1/test \
  --target_variables Erlotinib
```

`--pretrained_model`: Path to the trained model .pth file.

`--artifacts`: Path to the preprocessing artifacts (.joblib) saved during training.

`--data_path_test`: Folder containing the test dataset (same omics/features as training).

`--target_variables`: Target variable(s) to predict in the test set.

This works with all model classes including DirectPred, supervised_vae, CrossModalPred, MultiTripletNetwork, and GNN.

**Comment by amitpande74** [comment]
Thanks a lot @borauyar ! Learnt a lot during the process from you.
